### PR TITLE
Static Site generator 

### PR DESCRIPTION
**Pull request by Apoorv-Mittal**
This implementation is to try and address the issue #3. I have used Gatsby to create static website from the given and is renders completely identical website right now. I will need help implementing all the functionality. 
To start just do `yarn install` and then start the server by running `gatsby develop`.

### CODE REVIEW

File: package-lock.json [hunk 1]
```diff
@@ -0,0 +1,13230 @@
+{
```
Comment (philipp-spiess) [inline package-lock.json]: You can safely delete this file since we're using the yarn lockfile instead.

**Review by philipp-spiess — COMMENTED** [review]
Thanks @Apoorv-Mittal this already looks great. Before we merge this I have to make a few changes though since right now GitHub Pages expects an `index.html` file in the root of the document and I need to change that so that I can deploy the build artifacts on a separate branch. I also want to check if we can remove `gatsby-browser.js`, `gatsby-config.js`, or `gatsby-node.js` but I have to do some reading first. 

If you have the small checkmark in the PR view enabled that says that maintainers can push into your branch, I'll fix it up directly before I merge it. I'll try to do it later this week (I'm pretty busy today, sorry).

Thanks again!

File: package-lock.json (hunk 1 above)
Comment (Apoorv-Mittal) [reply package-lock.json]: Okay, I have removed this

**Comment by Apoorv-Mittal** [comment]
Yes, there is a check mark enabling you to add commits to my branch.

**Comment by philipp-spiess** [comment]
Thanks again for working on the Gatsby setup. I looked into it a bit and it seems that Gatsby is pretty powerful but also more complicated to use. For example the website does not seem to be server side rendered properly - the `<title/>` tag is empty, for example.

I'm not sure how to fix this and don't have the time right now to look into it unfortunately (a quick Google search brought me to the `gatsby-plugin-react-helmet` plugin which you've already added. No idea how to make it work from there).

All in all I'm not sure if Gatsby is the right choice for this project. I've worked with Next.js before and it seems to be a lot more lightweight which is probably a better fit for this project. I quickly checked out React Static as well and it seems to be more lightweight as well. I think I'll rather go with one of those to packages. 

And yes, I'm very very sorry but this means that I won't merge this pull request for now. 😞 I'm really really thankful for your work on this and I've also checked to make sure it's still counted for your Hacktoberfest contributions 

> **Do PRs have to be accepted/merged?**
> No. They will still count unless they are spam, irrelevant, or tagged as invalid.

https://hacktoberfest.digitalocean.com/faq

Maybe I'm seeing something wrong though. I'm just saying that I'm not fully sold on Gatsby for this specific project yet (I know that it is great for so may other projects)!

Thank you again for your help and sorry that I won't be accepting this for now.

**Comment by Apoorv-Mittal** [comment]
Thank you @philipp-spiess I completely understand what you are saying and I have to agree with you that Gatsby is not something which would be ideal for this project. I planned to go with Gatsby because I wanted to learn it and I had a lot about it. It might be best to use Next.js as it is very light weight and you already have experience with it.  Maybe if you want I can work on static site generation using Next.js and try make a PR this weekend.

Thank you for giving me the opportunity to work on this project.

**Comment by philipp-spiess** [comment]
@Apoorv-Mittal Thanks for your understanding.

To be honest I'm not sure about Next.js yet. I was looking at this "basic" example of React Static and it seems really well done. There's a [config file](https://github.com/nozzle/react-static/blob/master/examples/basic/static.config.js) which basically returns an object describing every page that should be generated meaning that I could dynamically parse a bunch of files from a folder (the individual newsletter issues) and then generate whatever I want from that. I really like that because it feels like there's no magic involved. What do you think about this? Check out [the basic example](https://github.com/nozzle/react-static/tree/master/examples).

On the other hand Next.js will automatically split every individual pages into their own files which will reduce the data payload to the bare minimum. IIRC the "default" way to use Next.js is that every page has it's own file. In order to have "dynamic" contents created from external files on the file system, I used a pre build step that was creating those Next.js pages for me. Worked fine for my use case (we plan to open source the implementation once the product is released 🙂).

**Comment by Apoorv-Mittal** [comment]
I think that using React Static might be the way to go because of the way it works. It will be easier for you to just put a new issue in the issue folder and then let React Static handle it because you have already predefined how you render the issues. It will be time efficient for you. Maybe making a new issue can be made a lot easier by writing a markdown for every issue. You can use the same markdown for a predefined email letter format as well. What do you think about this?

**Comment by philipp-spiess** [comment]
Yeah I was thinking the same. Wanna give it a shot? 🙂

**Comment by Apoorv-Mittal** [comment]
I will be happy too. I will make a PR soon 👍

**Comment by philipp-spiess** [comment]
Let's close this in favor of #10

**Pull request closed without merging**